File: Data Structures & Algorithms/binary-search/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let left = 0;
        let right = nums.length - 1;

        while (left<=right){
            //on veux le milieu index
            let milieux_index = Math.floor((left+right) /2)

            //si la valeur de target == celle de milieux return target
            if (target == nums[milieux_index]) return milieux_index;

            //si la valeur de target est plus petite que celle du milieu on garde left et right devient milieu
            if (target < nums[milieux_index]){
                right = milieux_index - 1;
            }

            else{
                left = milieux_index + 1;
            }

            //si la valeur de target est plus grasnde que celle du milieu on garde right et left devient milieu


        }

        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dict = new Dictionary<int, int>(); // valeur, index

       for (int i=0; i<nums.Length; i++)
       {
        var complement = target - nums[i];

        if (dict.ContainsKey(complement))
        {
            return new int[] {dict[complement], i};
        }

        dict[nums[i]] = i;

       }

       return new int[] {};
    }
}
